### ISSUE TITLE
Polybar loads wrong font

### ISSUE THREAD

**Issue by taketwo** [jaagr/polybar#1222]
I know there is a Wiki page and also dozens of issues regarding font loading, but after screening all of that I believe my case was not covered yet.

I want to use Font Awesome 5 icons, so in my config I have:

```ini
font-0 = FantasqueSansMono:pixelsize=10:bold:antialias=true;2
font-1 = "Font Awesome 5 Free:pixelsize=10:style=Regular;2"
```

The font is installed and can be matched with this name:

```bash
$ fc-list | grep "Font Awesome 5 Free"
/home/sergey/.fonts/nix-fonts/opentype/Font Awesome 5 Free-Solid-900.otf: Font Awesome 5 Free,Font Awesome 5 Free Solid:style=Solid,Book
/home/sergey/.fonts/nix-fonts/opentype/Font Awesome 5 Free-Regular-400.otf: Font Awesome 5 Free,Font Awesome 5 Free Regular:style=Regular
$ fc-match "Font Awesome 5 Free"
Font Awesome 5 Free-Regular-400.otf: "Font Awesome 5 Free" "Regular"
```

However, when I launch Polybar with this config and `-l info`, I get the following output:

```
* Loaded font "FantasqueSansMono:pixelsize=10:bold:antialias=true" (name=Fantasque Sans Mono, offset=2, file=/home/sergey/.fonts/nix-fonts/opentype/FantasqueSansMono-Bold.otf)
* Loaded font "Font Awesome 5 Free:pixelsize=10:style=Regular" (name=DejaVu Sans, offset=2, file=/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf)
```

For some reason, Polybar loads DejaVu instead, but believes it loaded the correct one. ([Here](https://pastebin.com/34z6cDtT) is the relevant part of the output with `FC_DEBUG=1`.)

I tried different variations of the font name (with and withou quotes, with and without `style=Regular` part), but none worked in Polybar, though each was possible to `fc-match`.

Any suggestions?

**Comment by patrick96** [comment]
Interesting case. What does `fc-match "Font Awesome 5 Free:pixelsize=10:style=Regular"` give you?
Can you also try `font-1 = Font Awesome 5 Free Regular:style=Regular`

**Comment by taketwo** [comment]
It matches correctly:

```bash
$ fc-match "Font Awesome 5 Free:pixelsize=10:style=Regular"
Font Awesome 5 Free-Regular-400.otf: "Font Awesome 5 Free" "Regular"
```

I tried to use your proposed variant, but it does not change anything, still loads DejaVu instead. In fact, whatever random string I put there, it reports:

```
* Loaded font "foobar" (name=DejaVu Sans, offset=0, file=/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf)
```

**Comment by patrick96** [comment]
Hm, so it looks like polybar and fc-match don't use the exact same way to find fonts. I think the problem is that your fonts are in your home directory. For some reason fc-list also looks in your `~/.fonts` directory. According to the [Arch Wiki](https://wiki.archlinux.org/index.php/font_configuration#Font_paths) `~/.fonts/` is deprecated, so maybe that's it. What distro are you using?
Also what is the output of `xset q`?

**Comment by taketwo** [comment]
I'm on Ubuntu, but extensively use [Nix package manager](https://en.wikipedia.org/wiki/Nix_package_manager) to get up-to-date apps and libraries. My fonts are also managed by Nix, though it puts installed `otf` files in some obscure location, so I created `$HOME/.fonts` and made a symlink inside. I recognize that this is a shady setup, but the other font (`font-0` in my config) is also installed through Nix and is located in exactly the same directory as Font Awesome, so that should not be a problem.

```bash
$ xset q
Keyboard Control:
  auto repeat:  on    key click percent:  0    LED mask:  00000002
  XKB indicators:
    00: Caps Lock:   off    01: Num Lock:    on     02: Scroll Lock: off
    03: Compose:     off    04: Kana:        off    05: Sleep:       off
    06: Suspend:     off    07: Mute:        off    08: Misc:        off
    09: Mail:        off    10: Charging:    off    11: Shift Lock:  off
    12: Group 2:     off    13: Mouse Keys:  off
  auto repeat delay:  660    repeat rate:  25
  auto repeating keys:  00ffffffdffffbbf
                        fadfffefffedffff
                        9fffffffffffffff
                        fff7ffffffffffff
  bell percent:  50    bell pitch:  400    bell duration:  100
Pointer Control:
  acceleration:  2/1    threshold:  4
Screen Saver:
  prefer blanking:  yes    allow exposures:  yes
  timeout:  0    cycle:  0
Colors:
  default colormap:  0x20    BlackPixel:  0x0    WhitePixel:  0xffffff
Font Path:
  /usr/share/fonts/X11/misc,/usr/share/fonts/X11/Type1,built-ins
DPMS (Energy Star):
  Standby: 0    Suspend: 0    Off: 0
  DPMS is Enabled
  Monitor is On
```

**Comment by taketwo** [comment]
OK, for what it's worth, I tried to move `$HOME/.fonts` into `$HOME/.local/share/fonts` and re-run `fc-cache -r`. Polybar picks up the font now!

Thanks for your help!

**Comment by patrick96** [comment]
still strange that it picked up your first font but not font awesome, even though they are in the same folder.

**Comment by Yves0** [comment]
It happened to me too. After installing a new Font 'Source_Code_Pro_Yahei-Hybrid.ttf', polybar loads it though I did not configure to use.Then poly bar performs strange..![image](https://user-images.githubusercontent.com/17212864/52206709-0dab4d80-28b6-11e9-857c-0b512c87c47b.png).
I tried to switch font-0 and font-1 then polybar pickup the right font now.
![image](https://user-images.githubusercontent.com/17212864/52247655-a0cf9c00-2925-11e9-8c66-06d72b7c2142.png)
